Update to doc for the new on-chain architecture (#728)
LiskHQ/lisk-docs: modules/ROOT/pages/accounts.adoc

= Accounts

> :description: This section explains the different account properties and how these relate to different transactions.

Reviewer: ```suggestion
:description: This section explains the different account properties and their correspondence to different modules.
```

== Account properties

> This section lists the properties stored in a Lisk account. Except for `address`, they are contained in four property modules: `token`, `sequence`, `keys`, and `dpos`.

Reviewer: I would say a module is the transaction schemas, account schema, transaction validation and logic together. So I would  say that the different objects in an account correspond to modules rather than calling these objects "property modules". But maybe Shu could give feedback on that.

> This section lists the properties stored in a Lisk account. Except for `address`, they are contained in four property modules: `token`, `sequence`, `keys`, and `dpos`.

Author: @shuse2 considering Jan's comment, what would you think is the proper way to refer to the account objects for the different modules? Are you ok with "object that correspond to a default module"?

> This section lists the properties stored in a Lisk account. Except for `address`, they are contained in four property modules: `token`, `sequence`, `keys`, and `dpos`.

Reviewer 2: yes i think 
> different objects, each corresponding to one of the four default modules: `token`, `sequence`, `keys`, and `dpos`.

make sense. maybe 
```
each corresponding to one of the account schema defined in four default modules
```
is slightly more descriptive?

> This section lists the properties stored in a Lisk account. Except for `address`, they are contained in four property modules: `token`, `sequence`, `keys`, and `dpos`.

Reviewer: Sounds good. Just small grammar fix
> each corresponding to one of the account schema**s** defined in **the** four default modules

=== `balance`

> The balance is an integer value indicating how many tokens are stored in the account and available to be used. It belongs to the `token` module of the account.

Reviewer: ```suggestion
The balance is an integer value indicating how many tokens are stored in the account and available to be used. It belongs to the `token` object of the account.
```

Reviewer: In JSON you call a collection of properties (key-value pairs) object. I am unsure whether it makes sense to say "module of the account".

The total number of tokens owned by an account is the sum of the balance and the tokens locked into <<votes,`votes`>> and <<unlocking,`unlocking`>>.
The balance can be sent from one account to another using a xref:{url_transactions_balance_transfer}[token transfer transaction].

=== `keys`

By default, a transaction issued by an account has to be signed with the private key corresponding to the original public key that generated the account address.
However, this default configuration can be changed, and issuing a transaction can require signatures by multiple different private keys (and not necessarily the default one).
The corresponding public keys are stored in the `keys` property.
These public keys can be either [#index-mandatory-1]#mandatory# or [#index-optional-1]#optional#: All mandatory keys have to sign the transaction, and the total number of signatures (mandatory plus optional) has to be equal to `numberOfSignatures` (also stored in `keys`).
Users can update the `keys` property with a xref:{url_transactions_multisignature}[multisignature group registration transaction], specifying the mandatory and optional keys and the number of required signatures.

=== `nonce`

> This property stores the current xref:{url_transactions_nonce}[*`nonce`*] of the account. It belongs to the `sequence` module of the account.

Reviewer: ```suggestion
This property stores the current xref:{url_transactions_nonce}[*`nonce`*] of the account. It belongs to the `sequence` object of the account.
```

Starting from zero, each transaction sent from the account increases the `nonce` value by one.
Hence the current `nonce` corresponds to the number of transactions sent from the account.

=== `dpos`

> It contains all the information relevant to the specific DPoS protocol of the chain.

Reviewer: ```suggestion
The `dpos` object contains all account properties related to the DPoS protocol.
```

Reviewer: As we don't use the account asset as container for sidechain-specific properties, I would not add "of the chain". Otherwise, we could add the same to the balance or keys object.

> The following properties are part of the `dpos` module in the default Lisk protocol.

Reviewer: ```suggestion
The following properties are part of the `dpos` object in the default Lisk protocol.
```

==== `unlocking`

> Tokens used to vote for a delegate are frozen, and have to be unlocked to make them available again. Similarly to the `votes` property, this property stores the voted-delegate's address and the amount, but also the block in which the votes have been removed from the delegate.

Reviewer: ```suggestion
Tokens used to vote for a delegate are locked, and have to be unlocked to make them available again. Similarly to the `votes` property, this property stores the voted-delegate's address and the amount, but also the block in which the votes have been removed from the delegate.
```

Reviewer: I would not use synonyms here as it can rather lead to confusion. `frozen` may sound indefinitely.

> After the appropriate waiting time, the user can then unlock the tokens with an xref:{url_transactions_unlock}[token unlock transaction].

Reviewer: ```suggestion
After the appropriate waiting time, the user can then unlock the tokens with a xref:{url_transactions_unlock}[token unlock transaction].
```
